modules/tensor_mechanics/src/materials/cohesive_zone_model/SlipWeakeningFriction2d.C

```diff
@@ -0,0 +1,163 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
```

Comment:
Can you make a slipWeakening base class with the common functions that the 2d and 3d models can derive off?

Reply:
Hi @lynnmunday, if I understand correctly, I should make a "SlipWeakeningBase" inherits from "CZMComputeLocalTractionTotalBase", and then two functions "SlipWeakeningFriction2d" and "SlipWeakeningFriction3d" inherits from "SlipWeakeningBase"?

Reply:
@chunhuizhao478 Can you run a 2D simulation using the 3D object or will the code crash?  Is there something you could initialize to 1 or 0 in the 3D object to make it work for a 2D simulation?

Reply:
Oh I see what you mean, yes I could close all the quantities related to the dip direction for 2D, and also adjust the nodal area and nodal mass accordingly, other than that I think the two versions can be merged. So is there a way to switch the case based on the input dimension?

Reply:
> Oh I see what you mean, yes I could close all the quantities related to the dip direction for 2D, and also adjust the nodal area and nodal mass accordingly, other than that I think the two versions can be merged. So is there a way to switch the case based on the input dimension? @lynnmunday

Reply:
You could add a parameter enum for 2D or 3D and make it required.